lib/client.js

```diff
@@ -41,6 +41,14 @@ function sendPacket (client, packet, cb) {
   }
 }
 
+function storePacket (client, packet, cb) {
+  client.outgoingStore.put(packet, function storedPacket (err) {
+    if (err) {
+      return cb && cb(err)
+    }
+  })
```

Comment:
I think we shouldn't, I believe caller assumes `cb` as a (un)successful send operation, however storing a packet for later processing is an inflight state, caller should be suspended.
in a non-crash scenario,  `cb` will be called later when connection is ready and packet is sent. (if we'd called it before, it will be called twice!)